dom/src/main/scala-3/fs2/dom/Dom.scala

```diff
@@ -48,6 +52,163 @@ object Node {
       node.replaceChild(newChild, oldChild)
       ()
     }
+
+    def addEventListener[T <: Event](`type`: String, options: EventListenerOptions)(using
+        F: Dom[F]
+    ): Resource[F, Stream[F, T]] =
+      EventTargetHelpers.listen(node, `type`, options)
+
+    def attributes(using F: Dom[F]): F[NamedNodeMap[F]] =
+      F.delay(node.attributes)
+
+    def nodeType: Int = node.nodeType
+
+    def previousSibling(using F: Dom[F]): F[Node[F]] =
+      F.delay(node.previousSibling)
+
+    def localName(using F: Dom[F]): F[String] = F.delay(node.localName)
+
+    def namespaceURI(using F: Dom[F]): F[Option[String]] = F.delay(Option(node.namespaceURI))
+
+    def textContent(using F: Dom[F]): F[String] = F.delay(node.textContent)
+
+    def setTextContent(s: String)(using F: Dom[F]): F[Unit] = F.delay(node.textContent = s)
```

Comment:
Instead of using setters/getters, I've been using `Ref`. Example:
https://github.com/armanbilge/fs2-dom/blob/8ff10b7eecf42ded294d0c5d85e7c876c4878ac6/dom/src/main/scala-3/fs2/dom/Dom.scala#L273-L277